### PR TITLE
Use std::strings for string2… helpers in key.cpp

### PR DESCRIPTION
**Pull request by dnnr**

### CODE REVIEW

**Comment by dnnr** [comment]
This includes #288

File: src/key.cpp [hunk 1]
```diff
@@ -103,51 +107,60 @@ int keybind(int argc, char** argv, Output output) {
...
+    for (auto &sep : KEY_COMBI_SEPARATORS) {
+        std::replace(keySpec.begin(), keySpec.end(), sep, baseSep);
+    }
+
+    // Split spec into tokens:
+    vector<std::string> tokens;
+    std::string token;
+    std::istringstream tokenStream(keySpec);
+    while (std::getline(tokenStream, token, baseSep)) {
+        tokens.push_back(token);
```
Comment (t-wissmann) [inline src/key.cpp]: Don't you need to strip off `baseSep` from the end of token?

Comment (dnnr) [reply src/key.cpp]: The delimiter gets dropped by `std::getline` ("the delimiter character is extracted from input, but is not appended to str." https://en.cppreference.com/w/cpp/string/basic_string/getline)

Comment (t-wissmann) [reply src/key.cpp]: Thanks!

**Pull request merged**